Discovery throwing exception

@craigham reported the following exception

```

2017-07-01 06:13:51.313 [DEBUG] [b.o.d.OmnilinkDiscoveryService:66 ] - Starting scan 2017-07-01 06:13:51.415 [ERROR] [.o.b.o.d.ObjectPropertyRequest:61 ] - Error retrieving object propertiescom.digitaldan.jomnilinkII.OmniInvalidResponseException: null at com.digitaldan.jomnilinkII.Connection.reqObjectProperties(Connection.java:538) at org.openhab.binding.omnilink.handler.OmnilinkBridgeHandler.reqObjectProperties(OmnilinkBridgeHandler.java:460) at org.openhab.binding.omnilink.discovery.ObjectPropertyRequest$1.computeNext(ObjectPropertyRequest.java:51) at org.openhab.binding.omnilink.discovery.ObjectPropertyRequest$1.computeNext(ObjectPropertyRequest.java:1) at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143) at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138) at org.openhab.binding.omnilink.discovery.OmnilinkDiscoveryService.discoverAreas(OmnilinkDiscoveryService.java:205) at org.openhab.binding.omnilink.discovery.OmnilinkDiscoveryService.startScan(OmnilinkDiscoveryService.java:70) at org.eclipse.smarthome.config.discovery.AbstractDiscoveryService.startScan(AbstractDiscoveryService.java:199) at org.eclipse.smarthome.config.discovery.internal.DiscoveryServiceRegistryImpl.startScan(DiscoveryServiceRegistryImpl.java:382) at org.eclipse.smarthome.config.discovery.internal.DiscoveryServiceRegistryImpl.startScans(DiscoveryServiceRegistryImpl.java:358)
```

--- a/addons/binding/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/discovery/ObjectPropertyRequests.java
+++ b/addons/binding/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/discovery/ObjectPropertyRequests.java
@@ -22,16 +22,16 @@ public class ObjectPropertyRequests<T extends ObjectProperties> {
             Message.OBJ_TYPE_CONSOLE, ConsoleProperties.class);
 
     public final static ObjectPropertyRequests<AreaProperties> AREA = new ObjectPropertyRequests<>(
-            Message.OBJ_TYPE_CONSOLE, AreaProperties.class);
+            Message.OBJ_TYPE_AREA, AreaProperties.class);
 
     public final static ObjectPropertyRequests<ZoneProperties> ZONE = new ObjectPropertyRequests<>(
-            Message.OBJ_TYPE_CONSOLE, ZoneProperties.class);
+            Message.OBJ_TYPE_ZONE, ZoneProperties.class);
 
     public final static ObjectPropertyRequests<UnitProperties> UNIT = new ObjectPropertyRequests<>(
-            Message.OBJ_TYPE_CONSOLE, UnitProperties.class);
+            Message.OBJ_TYPE_UNIT, UnitProperties.class);
 
     public final static ObjectPropertyRequests<AudioZoneProperties> AUDIO_ZONE = new ObjectPropertyRequests<>(
-            Message.OBJ_TYPE_CONSOLE, AudioZoneProperties.class);
+            Message.OBJ_TYPE_AUDIO_ZONE, AudioZoneProperties.class);
 
     private final int propertyRequest;
     private final Class<T> responseType;

--- a/addons/binding/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/discovery/OmnilinkDiscoveryService.java
+++ b/addons/binding/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/discovery/OmnilinkDiscoveryService.java
@@ -254,11 +254,12 @@ public class OmnilinkDiscoveryService extends AbstractDiscoveryService {
                     .builder(bridgeHandler, ObjectPropertyRequests.UNIT).selectNamed().areaFilter(areaFilter)
                     .selectAnyLoad().build();
 
+            int currentRoom = 0;
+            String currentRoomName = "";
+
             for (UnitProperties unitProperties : objectPropertyRequest) {
 
                 int objnum = unitProperties.getNumber();
-                int currentRoom = 0;
-                String currentRoomName = "";
 
                 // boolean isInRoom = false;
                 boolean isRoomController = false;

